src/graphmol/rw_mol.rs

```diff
@@ -48,11 +58,23 @@ impl RWMol {
         ROMol { ptr }
     }
 
-    pub fn from_smarts(smarts: &str) -> Result<Self, Box<dyn std::error::Error>> {
+    pub fn from_smarts(smarts: &str) -> Result<Self, RWMolError> {
+        if smarts.is_empty() {
+            return Err(RWMolError::EmptyInputError);
+        }
```

Comment:
Let's remove this check and enum variant. We can leave empty string checking to users or rdkit.